local-volume/bootstrapper/README.md

```diff
@@ -1,9 +1,85 @@
+local-volume-provisioner-bootstrap is used to bootstrap provisioner. The main use
+case of bootstrapper is to make provisioner configurable. Below is a detailed flow
+of how the bootstrap process works:
+
+- looks for a configmap passed in via flag `-volume-config`; otherwise, a default
+  configmap named `local-volume-default-config` will be created
+- reads and validates the configmap, then auto-generates missing configurations,
+  see below
+- looks for a service account passed in via flag `-serviceaccount`; otherwise, a
+  default service account named `local-storage-admin` will be created
+- creates two cluster role bindings for the service account, i.e. `system:persistent-volume-provisioner`
+  and `system:node`. The role bindings are named `local-storage:provisioner-node-binding` and
+  `local-storage:provisioner-pv-binding` respectively.
+- creates provisioner daemonset based on the service account, volume configmap,
+  `-image` option, etc
+- exit
+
+## Configuration
+
+### Volume config
+
+Volume config is a map from storage class to volume configuration, i.e. `map[string]MountConfig`,
+where `MountConfig` is the configuration for a single volume, see below.
+
+```go
+type MountConfig struct {
+	// The hostpath directory
+	HostDir string `json:"hostDir"`
+	// The mount point of the hostpath volume
+	MountDir string `json:"mountDir"`
+}
+```
+
+- `HostDir` is required; it points to the directory where provisioner looks for
+  local volumes.
+- `MountDir` is optional, it is the path inside container where `hostDir` is mounted
+  to. If omitted, `MountDir` will be auto-generated by bootstrapper. The generation
+  rule is to trim '/' prefix and change "/" to "~" (based on kubernetes convention),
+  then concatenate with root path. For example, suppose `-mountRoot` flag equals to
+  "/mnt/local-storage" and `hostDir` equals to "/mnt/others", then generated `MountDir`
+  will be "/mnt/local-storage/mnt~others".
+
+Below is an example configmap:
+
+```yaml
+kind: ConfigMap
+metadata:
+  name: local-volume-config
+  namespace: kube-system
+data:
+  "local-fast": |
+    {
+      "hostDir": "/mnt/ssds",
+      "mountDir": "/local-ssds"
+    }
+  "local-slow": |
+    {
+      "hostDir": "/mnt/hdds",
+      "mountDir": "/local-hdds"
+    }
+  "local-storage": |
+    {
+      "hostDir": "/mnt/disks"
+    }
+```
+
+### Command line options
+
+To see all options, compile bootstrapper and use `-h` option, below is a curated
+list of important options:
+
+- `-image`: Name of local volume provisioner image (default "quay.io/external_storage/local-volume-provisioner")
+- `-volume-config`: Name of the local volume configuration configmap. The configmap
+  must reside in the same namespace with bootstrapper. (default "local-volume-default-config")
+- `-serviceaccount`: Name of the service accout for local volume provisioner
+  (default "local-storage-admin")
+- `-mount-root`: Container root directory of volume mount path for discoverying
```

Comment:
Not quite necessary, but i'm not sure if people want to change it... Since it's already there, I'll just remove the description from README.